File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(index, total):
        
        
        if index == len(numbers):
            return 1 if total == target else 0
        
        plus = dfs(index + 1, total + numbers[index])
        minus = dfs(index + 1, total - numbers[index])
        return plus + minus  # 여기서 결과값들이 합산됩니다!
    
    return dfs(0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let count = 0;
    
    function dfs(index, total) {
        if (index === numbers.length) {
            if (total === target) {
                count++;
            }
            return;
        }
        
        dfs(index + 1, total + numbers[index]);
        dfs(index + 1, total - numbers[index]);
    }
    
    dfs(0, 0);
    return count;
}